### PR TITLE
fix(xdsl.templates): display error reason on template creation

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #UXCT-592
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~Only FR translations have been updated~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~Breaking change is mentioned in relevant commits~

## Description

Display reason of the error on template creation

### CODE REVIEW

**Review by antonymarion — APPROVED** [review]

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/modem/templates/pack-xdsl-modem-templates.controller.js [hunk 1]
```diff
@@ -92,6 +92,9 @@ export default class XdslModemTemplateCtrl {
           type: 'error',
           detail: errorMessage,
         };
+        if (err.data.message.includes(this.TEMPLATE_CONSTANT.errors.modemWifi)) {
+          this.message.reason = err.data.message;
+        }
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/xdsl/modem/templates/pack-xdsl-modem-templates.controller.js:97]:
```suggestion
        if (err.data?.message.includes(this.TEMPLATE_CONSTANT.errors.modemWifi)) {
          this.message.reason = err.data?.message;
        }
```

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/modem/templates/pack-xdsl-modem-templates.html [hunk 2]
```diff
@@ -10,11 +10,15 @@
         <oui-message
             type="success"
             data-ng-if="$ctrl.message.type === 'success'"
-            >{{$ctrl.message.detail}}</oui-message
-        >
-        <oui-message type="error" data-ng-if="$ctrl.message.type === 'error'"
-            >{{$ctrl.message.detail}}</oui-message
         >
+            {{$ctrl.message.detail}}
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/xdsl/modem/templates/pack-xdsl-modem-templates.html:14]:
```suggestion
<span data-ng-bind="$ctrl.message.detail"></span>
```

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/modem/templates/pack-xdsl-modem-templates.html [hunk 3]
```diff
@@ -10,11 +10,15 @@
...
+        </oui-message>
+        <oui-message type="error" data-ng-if="$ctrl.message.type === 'error'">
+            <div>{{$ctrl.message.detail}}</div>
+            <div data-ng-if="$ctrl.message.reason">
+                {{$ctrl.message.reason}}
+            </div>
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/xdsl/modem/templates/pack-xdsl-modem-templates.html:20]:
```suggestion
            <p data-ng-bind="$ctrl.message.detail" ></p>
            <p data-ng-if="$ctrl.message.reason" data-ng-bind="$ctrl.message.reason">
            </p>
```

**Review by antonymarion — DISMISSED** [review]

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/modem/templates/pack-xdsl-modem-templates.html (hunk 2 above)
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/pack/xdsl/modem/templates/pack-xdsl-modem-templates.html:14]: You have done for error message but the modification isn't done for success message. you have to use data-ng-bind

**Review by ghyenne — CHANGES_REQUESTED** [review]
You have done for error message but the modification isn't done for success message. you have to use data-ng-bind

**Review by ghyenne — DISMISSED** [review]

**Review by mhelhali-soufien — APPROVED** [review]

**Review by antonymarion — APPROVED** [review]

**Pull request merged**